internal/host/static/repository_host.go

```diff
@@ -22,45 +22,50 @@ import (
 // Both h.Name and h.Description are optional. If h.Name is set, it must be
 // unique within h.CatalogId.
 func (r *Repository) CreateHost(ctx context.Context, scopeId string, h *Host, opt ...Option) (*Host, error) {
+	const op = errors.Op("static.CreateHost")
 	if h == nil {
-		return nil, fmt.Errorf("create: static host: %w", errors.ErrInvalidParameter)
+		return nil, errors.New(errors.InvalidParameter, errors.WithOp(op), errors.WithMsg("nil host"))
 	}
 	if h.Host == nil {
-		return nil, fmt.Errorf("create: static host: embedded Host: %w", errors.ErrInvalidParameter)
+		return nil, errors.New(errors.InvalidParameter, errors.WithOp(op), errors.WithMsg("nil embedded host"))
 	}
 	if h.CatalogId == "" {
-		return nil, fmt.Errorf("create: static host: no catalog id: %w", errors.ErrInvalidParameter)
+		return nil, errors.New(errors.InvalidParameter, errors.WithOp(op), errors.WithMsg("no catalog id"))
 	}
 	if h.PublicId != "" {
-		return nil, fmt.Errorf("create: static host: public id not empty: %w", errors.ErrInvalidParameter)
+		return nil, errors.New(errors.InvalidParameter, errors.WithOp(op), errors.WithMsg("public id not empty"))
 	}
 	if scopeId == "" {
-		return nil, fmt.Errorf("create: static host: no scopeId: %w", errors.ErrInvalidParameter)
+		return nil, errors.New(errors.InvalidParameter, errors.WithOp(op), errors.WithMsg("no scope id"))
 	}
 	h.Address = strings.TrimSpace(h.Address)
 	if len(h.Address) < MinHostAddressLength || len(h.Address) > MaxHostAddressLength {
-		return nil, fmt.Errorf("create: static host: bad address: %w", ErrInvalidAddress)
+		return nil, errors.New(errors.InvalidAddress, errors.WithOp(op))
 	}
 	h = h.clone()
 
 	opts := getOpts(opt...)
 
 	if opts.withPublicId != "" {
 		if !strings.HasPrefix(opts.withPublicId, HostPrefix+"_") {
-			return nil, fmt.Errorf("create: static host: passed-in public ID %q has wrong prefix, should be %q: %w", opts.withPublicId, HostPrefix, errors.ErrInvalidPublicId)
+			return nil, errors.New(
```

Comment:
Just an observation, that this change may require a change to the caller, if they used `errors.Is(err, errors.ErrInvalidPublicId)` to determine how to handle the error.   If you wrapped the sentinel `errors.ErrInvalidPublicId` then you'd be backward compatible.... if that's important.

Reply:
I could not find anywhere in the code that the sentinels removed in this PR are checked (other than some tests not included in this path), perhaps I missed something.